### DISCUSSION TITLE
--components flag for k8s removed in 0.58

### DISCUSSION THREAD

**Discussion by weisinc** [aquasecurity/trivy#8131] [Q&A]
### Question

Is the --components flag in k8s removed from 0.58 ? I often use it to avoid creating a default namespace for "infra" scanning in the cluster. What is the workaround to avoid creating namespace to scan a cluster without the --components workload option ?

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

None

### Mode

Standalone

### Operating System

macOS Sequoia v15.2

### Version

```bash
% trivy -v
Version: 0.58.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-12-18 18:17:47.633228986 +0000 UTC
  NextUpdate: 2024-12-19 18:17:47.633228596 +0000 UTC
  DownloadedAt: 2024-12-18 18:45:35.62208 +0000 UTC
Check Bundle:
  Digest: sha256:27b408ac9a1d8d6a984bd98a2062792afa58d4534b8fa2dc46398cc6cd6a738a
  DownloadedAt: 2024-12-18 18:47:55.386239 +0000 UTC
```

**Comment by itaysk** [comment]
> Is the --components flag in k8s removed from 0.58 ?
the kubernetes command experience was [changed in v0.52](https://github.com/aquasecurity/trivy/discussions/6622). you can see a summary of changes in the linked release notes and details in the documentation. 

> avoid creating a default namespace for "infra"
I'm not sure if I understand what you mean with the default namespace, but perhaps you want to `--exclude-namespaces kube-system`
